### PR TITLE
Flag submission process errors

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

This PR addresses issue #107 

## Sneak peek

screenshot-2018-1-10 submissions

### CODE REVIEW

File: apps/submission/models.py [hunk 1]
```diff
@@ -61,3 +62,19 @@ def archive_upload_to(instance, filename):
     )
 
     imported = models.BooleanField(default=False)
+
+    @property
+    def is_failed(self):
```
Comment (willdurand) [inline apps/submission/models.py]: `has_failed` maybe?

File: apps/submission/models.py [hunk 2]
```diff
@@ -61,3 +62,19 @@ def archive_upload_to(instance, filename):
...
+        """Check if process has failed tasks"""
+
+        if self.task_set.filter(status=STATUS.ERROR).count() > 0:
```
Comment (willdurand) [inline apps/submission/models.py]: So we do not set any `status` to the main object when one of the tasks errored? Like it is done for `DONE` after.

File: apps/submission/models.py [hunk 3]
```diff
@@ -61,3 +62,19 @@ def archive_upload_to(instance, filename):
...
+            return True
+        return False
+
+    @property
+    def is_done(self):
+        """Check if the process is done"""
+
+        if self.status == STATUS.DONE:
```
Comment (willdurand) [inline apps/submission/models.py]:
```
return self.status == STATUS.DONE
```

File: apps/submission/models.py (hunk 2 above)
Comment (jmaupetit) [reply apps/submission/models.py]: Unfortunately no. I had the same expectation, but, after reading viewflow's code again and again, it appears that it has been designed this way, _i.e._ a task error is never propagated to its process. Odd.

**Pull request merged**